### PR TITLE
Embed SonarC# 8.53 and SonarVB 8.53

### LINKED ISSUE #3752: Embed SonarC# 8.53 and SonarVB 8.53
Release notes: https://github.com/SonarSource/sonar-dotnet/releases/tag/8.53.0.62665

3 FP, one of which (https://github.com/SonarSource/sonar-dotnet/issues/6616) is a fix for a code fix on the `is` operator.
3 Improvements, one of which (https://github.com/SonarSource/sonar-dotnet/issues/6321) is a fix on highlighted content.
6 new rules.

### PR DESCRIPTION
**Pull request by duncanp-sonar**
Fixes #3752

### CODE REVIEW

File: src/Integration.Vsix/AsmRef_Integration.Vsix_Baseline_WithStrongNames.txt [hunk 1]
```diff
@@ -165,14 +165,17 @@ Relative path: 'SonarLint.VisualStudio.ConnectedMode.dll'
 
 Referenced assemblies:
 - 'LibGit2Sharp, Version=0.26.0.0, Culture=neutral, PublicKeyToken=7cbde695407f0333'
+- 'Microsoft.VisualStudio.ComponentModelHost, Version=17.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
+- 'Microsoft.VisualStudio.Shell.15.0, Version=17.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
+- 'Microsoft.VisualStudio.Shell.Framework, Version=17.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'
```
Comment (duncanp-sonar) [inline src/Integration.Vsix/AsmRef_Integration.Vsix_Baseline_WithStrongNames.txt:170]: Not sure why these changes are here. Result of previous incorrect merge maybe?

**Comment by duncanp-sonar** [comment]
@rita-sonarsource let's discuss tomorrow whether to include this analyzer update in the release (looks ok, and I did a smoke-test)

File: src/Integration.Vsix/AsmRef_Integration.Vsix_Baseline_WithStrongNames.txt (hunk 1 above)
Comment (duncanp-sonar) [reply src/Integration.Vsix/AsmRef_Integration.Vsix_Baseline_WithStrongNames.txt:170]: @rita-sonarsource the two added references are missing from the previous version of the file in `branch-sse-master3` - [here](https://github.com/SonarSource/sonarlint-visualstudio/blob/d05062f01d25347a302058f171c3a902b22f5ce9/src/Integration.Vsix/AsmRef_Integration.Vsix_Baseline_WithStrongNames.txt#L175).

I suspect it is the result of a bad merge. It won't cause a build failure in the PR CI run since that is not signing the assemblies -> the PR CI will be checking the other asmref baseline file (the one that contains the report for unsigned assemblies, which is correct).

To test, I've triggered a rebuild of the `branch-sse-master3` -> `master` PR with `ForceVsixSign` set to `true`. I think this will force the assemblies to be signed too, and I would expect the build to fail because the asmrefs are different.

Comment (rita-sonarsource) [reply src/Integration.Vsix/AsmRef_Integration.Vsix_Baseline_WithStrongNames.txt:170]: I ran rebuild with asmref and pushed an updated version to master3
